src/gsIO/gsWriteParaview.hpp

```diff
@@ -1660,6 +1661,144 @@ void gsWriteParaviewTrimmedCurve(const gsTrimSurface<T>& surf,
 
 }
 
+template<class T>
+void gsWriteParaview(const gsPolyField<T> & fields, std::string const & fn,
+                     unsigned npts, bool mesh, unsigned nptsMesh)
+{
+    gsParaviewCollection collection(fn);
+    gsParaviewCollection collectionMesh(fn + "_mesh");
+    std::string fileName = fn.substr(fn.find_last_of("/\\")+1);
+
+    for (index_t p = 0; p < fields.nPatches(); ++p)
+    {
+        //==== Can be moved to a separate funtion *gsWriteSinglePatchPolyField* ===//
+
+        const gsGeometry<> & geometry = fields.patch(p);
+        const unsigned d = geometry.domainDim();
+        GISMO_ASSERT(d <= 3, "Data has parametric dimension " + util::to_string(d) +
+                             ". Can't plot.\n");
+        const unsigned n = geometry.targetDim();
+        GISMO_ASSERT(n <= 3, "Data has dimension " + util::to_string(n) +
+                             ". Can't plot.\n");
+
+        gsVector<> a = geometry.support().col(0);
+        gsVector<> b = geometry.support().col(1);
+
+        gsVector<unsigned> np = uniformSampleCount(a,b,npts);
+        gsMatrix<> pts = gsPointGrid(a,b,np);
+
+        gsMatrix<> eval_geo;
+        geometry.eval_into(pts,eval_geo);
+        std::map<std::string, gsMatrix<> > data;
+        for (size_t i = 0; i < fields.nFields(); ++i)
+        {
+            fields.isParametric(i) ? fields.igaFunction(p,i).eval_into(pts,data[fields.name(i)])
+                                   : fields.function(p,i).eval_into(eval_geo,data[fields.name(i)]);
+
+            if (data[fields.name(i)].rows() == 2)
+            {
+                data[fields.name(i)].conservativeResize(3,eval_geo.cols() );
+                data[fields.name(i)].row(2).setZero();
+            }
+        }
+
+        if ( 3 - d > 0)
+        {
+            np.conservativeResize(3);
+            np.bottomRows(3-d).setOnes();
+        }
+        if ( 3 - n > 0 )
+        {
+            eval_geo.conservativeResize(3,eval_geo.cols() );
+            eval_geo.bottomRows(3-n).setZero();
+        }
+
+        gsWriteParaviewTPgrid(eval_geo, data, np.template cast<index_t>(), fn + util::to_string(p));
+
+        //=========================================================================//
+
+        collection.addPart(fileName + util::to_string(p),".vts");
+
+        if (mesh)
+        {
+            // if none of the fields is a BSpline function use basis of the geometry to plot mesh (may be coarse)
+            int index = -1;
+            for (size_t i = 0; i < fields.nFields(); ++i)
+                if (fields.isParametrized(i))
```

Comment:
What is if fields.isParametrized(i) holds for several i?

Reply:
A good question. I don't know. This is a problem with plotting meshes. Which one to plot? One of the geometry? One of the fields'? Or all of them somehow? The current solution to plot the first available field's mesh is not ideal, of course. 

Would you have any suggestions?